packages/blockchain-link/src/workers/blockbook/websocket.ts

```diff
@@ -222,20 +219,16 @@ export class BlockbookAPI extends TypedEmitter<BlockbookEvents> {
                 ...this.options.headers,
             },
         });
-        if (typeof ws.setMaxListeners === 'function') {
-            ws.setMaxListeners(Infinity);
-        }
         ws.once('error', error => {
             this.onClose();
             dfd.reject(new CustomError('websocket_runtime_error', error.message));
         });
         ws.on('open', () => {
+            this.ws = ws;
```

Comment:
Is there a reason for this change? I'm not saying it doesn't work, I'm just a bit paranoid.

Reply:
i think this is how it should be done from the beginning, websocket instance should be assigned **only** after successful connection.
there is a multiple functions below checking if `this.ws` is defined but it doesn't mean that `this.ws` is connected

Reply:
I've thought about all the possible scenarios a bit and it seems fine.

Reply:
Actually, it breaks the `Handle connection timeout` unit tests. So either this or the tests must be fixed.

Reply:
you're right! it did brake it (onTimeout > ws.listenerCount('open') > 0 condition) :+1: 

fixuped [b9a1a43](https://github.com/trezor/trezor-suite/pull/8718/commits/b9a1a435f6490660b64d0d9f983ef21a1cb4d217)